### PR TITLE
Suggest network-online.target for rc.local

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Review by bluca — APPROVED** [review]
lgtm

File: man/systemd-rc-local-generator.xml [hunk 1]
```diff
@@ -19,28 +19,40 @@
...
+    original System V version, which was executed "last" in the boot process, which is a concept that does
+    not translate to systemd.</para>
+
+    <para>Also note that <filename>rc-local.service</filename> is ordered after
+    <filename>network.target</filename>, which does not mean that the network is functional, see
+    <citerefentry><refentrytitle>systemd.special</refentrytitle><manvolnum>7</manvolnum></citerefentry>.
+    If the script requires a configured network connection, it may be desirable to pull in and order it after
+    <filename>network-online.target</filename> with:</para>
+
+    <programlisting>systemctl add-wants network-online.target rc-local.service</programlisting>
```
Comment (mbiebl) [inline man/systemd-rc-local-generator.xml]: Isn't this the wrong way around?
rc-local.service should have wants/after network-online.target, not network-online.target wanting rc-local.service?
```
# systemctl add-wants network-online.target rc-local.service 
Created symlink /etc/systemd/system/network-online.target.wants/rc-local.service → /lib/systemd/system/rc-local.service.
```

Comment (mbiebl) [reply man/systemd-rc-local-generator.xml]: `systemctl add-wants` doesn't add add a `After=` ordering, only a `Wants=` depdendency, right?

Comment (keszybz) [reply man/systemd-rc-local-generator.xml]: Yikes, thanks for catching that!

> `systemctl add-wants` doesn't add add a `After=` ordering, only a `Wants=` depdendency, right?

A symlink in  .wants/.requires implies also the ordering dependency After.

Comment (mbiebl) [reply man/systemd-rc-local-generator.xml]: > A symlink in .wants/.requires implies also the ordering dependency After.

Hm, I must be doing something wrong then:
```
# systemctl show -p Wants -p After rc-local.service
Wants=
After=basic.target sysinit.target system.slice systemd-journald.socket network.target

# systemctl add-wants rc-local.service network-online.target
Created symlink /etc/systemd/system/rc-local.service.wants/network-online.target → /lib/systemd/system/network-online.target.

# systemctl show -p Wants -p After rc-local.service
Wants=network-online.target
After=basic.target sysinit.target network.target systemd-journald.socket system.slice
```

Comment (keszybz) [reply man/systemd-rc-local-generator.xml]: Hmm. Indeed. I thought that this is the case, but the man page says only:
> target units will complement all configured dependencies of type Wants= or Requires= with dependencies of type After=

It seems @poettering thought so too:
> But if we ask people that, then we could as well ask them to instead do:
>
> mkdir -p /etc/systemd/system/rc-local.service.wants/
> ln -s /usr/lib/systemd/system/network-online.target /etc/systemd/system/rc-local.service.wants/
> 
> Which actually will do both: add Wants= + After=, and thus be more comprehensive.

Duh.

Comment (keszybz) [reply man/systemd-rc-local-generator.xml]: OK, I updated the man page to suggest a drop-in. It's not very nice, but I don't see a better approach.

**Pull request merged**